apps/builder/app/shared/style-object-model.test.ts

```diff
@@ -558,3 +588,252 @@ test("resolve non-cyclic references in custom properties", () => {
       .usedValue
   ).toEqual({ type: "keyword", value: "red" });
 });
+
+test("cascade value from active breakpoints", () => {
+  // @media small { body { width: 20px } }
+  // @media large { body { width: 30px } }
+  // @media base { body { width: 10px } }
+  const styles: StyleDecl[] = [
+    {
+      breakpointId: "small",
+      styleSourceId: "bodyLocal",
+      property: "width",
+      value: { type: "unit", value: 20, unit: "px" },
+    },
+    {
+      breakpointId: "large",
+      styleSourceId: "bodyLocal",
+      property: "width",
+      value: { type: "unit", value: 30, unit: "px" },
+    },
+    {
+      breakpointId: "base",
+      styleSourceId: "bodyLocal",
+      property: "width",
+      value: { type: "unit", value: 10, unit: "px" },
+    },
+  ];
+  const model: StyleObjectModel = {
+    styleSourcesByInstanceId: new Map([["body", ["bodyLocal"]]]),
+    styleByStyleSourceId: getStyleByStyleSourceId(styles),
+  };
+  const styleSelector: StyleSelector = {
+    instanceSelector: ["body"],
+    activeBreakpoints: ["base", "small", "large"],
+    activeStates: new Set(),
+  };
+  expect(
+    getComputedStyleDecl({ model, styleSelector, property: "width" }).usedValue
+  ).toEqual({ type: "unit", unit: "px", value: 30 });
+});
+
+test("ignore values from inactive breakpoints", () => {
+  // @media base { body { width: 10px } }
+  // @media small { body { width: 20px } }
+  // @media large { body { width: 30px } }
+  const styles: StyleDecl[] = [
+    {
+      breakpointId: "base",
+      styleSourceId: "bodyLocal",
+      property: "width",
+      value: { type: "unit", value: 10, unit: "px" },
+    },
+    {
+      breakpointId: "small",
+      styleSourceId: "bodyLocal",
+      property: "width",
+      value: { type: "unit", value: 20, unit: "px" },
+    },
+    {
+      breakpointId: "large",
+      styleSourceId: "bodyLocal",
+      property: "width",
+      value: { type: "unit", value: 30, unit: "px" },
+    },
+  ];
+  const model: StyleObjectModel = {
+    styleSourcesByInstanceId: new Map([["body", ["bodyLocal"]]]),
+    styleByStyleSourceId: getStyleByStyleSourceId(styles),
+  };
+  const styleSelector: StyleSelector = {
+    instanceSelector: ["body"],
+    // large is not active breakpoint
+    activeBreakpoints: ["base", "small"],
```

Comment:
I think the right term is "applicable", e.g. applicableBreakpoints

Reply:
This is more about whether provided list of breakpoints is activated from ui and rendered in canvas

Reply:
checked with the spec, that's the term they use

Reply:
> This is more about whether provided list of breakpoints is activated from ui and rendered in canvas

From builder standpoint yes, but from the logic that's the list of breakpoints you need to consider when calculating the value, right? 

I mean these breakpoints conditions are satisfying the viewport potentially, right?

Reply:
> checked with the spec, that's the term they use

please attach a link

Reply:
https://www.w3.org/TR/mediaqueries-5/ you will find the word "apply"  and "applicable"

Reply:
Very vague. Styles are applicable with certain condition condition. Conditions are not applicable, they define when to apply

Reply:
The query already applied/activated/matched

Reply:
then seems it is applicable :)

Reply:
applicable - can be potentially applied

here the query already active/matching

Reply:
I think applicable is when they are actually machtching, same semantics, but feel free to use "matching" ... active is just overused and often has too many meanings